arch/xtensa/src/esp32/esp32_rtc.c

```diff
@@ -2188,7 +2189,7 @@ int up_rtc_cancelalarm(enum alm_id_e alarmid)
 
   if (cbinfo->ac_cb != NULL)
     {
-      flags = spin_lock_irqsave(NULL);
+      flags = spin_lock_irqsave(&g_rtc_lock);
```

Comment:
```suggestion
      sched_lock();
      flags = spin_lock_irqsave(&g_rtc_lock);
```
 https://github.com/apache/nuttx/pull/15232 Could we merge similar commits into one commit? This makes reviewing easier.